doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1189,428 @@ processed in this order:
+configuration value overrides the manifest file for both groups.
+
+Therefore, projects ``foo`` and ``bar`` are both active.
+
+Example 7: Blocking multiple groups via configuration
+-----------------------------------------------------
+
+The entire manifest file is:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+       - name: bar
+         groups:
+           - groupA
+       - name: baz
+         groups:
+           - groupA
+           - groupB
+
+     defaults:
+       remote: example-remote
+     remotes:
+       - name: example-remote
+         url-base: https://git.example.com
+
+The ``manifest.groups`` configuration option is set to ``-groupA,-groupB``
+(you can ensure this by running ``west config manifest.groups --
+"-groupA,-groupB"``).
+
+In this case, both ``groupA`` and ``groupB`` are blocked.
+
+Therefore, projects ``foo`` and ``bar`` are both inactive.
+
+.. _west-manifest-submodules:
+
+Git Submodules in Projects
+**************************
+
+You can use the ``submodules`` keys briefly described :ref:`above
+<west-manifest-files>` to force ``west update`` to also handle any `Git
+submodules`_ configured in project's git repository. The ``submodules`` key can
+appear inside ``projects``, like this:
+
+.. code-block:: YAML
+
+   manifest:
+     projects:
+       - name: some-project
+         submodules: ...
+
+The ``submodules`` key can be a boolean or a list of mappings. We'll describe
+these in order.
+
+Option 1: Boolean
+=================
+
+This is the easiest way to use ``submodules``.
+
+If ``submodules`` is ``true`` as a ``projects`` attribute, ``west update`` will
+update the project's Git submodules whenever it updates the project itself. If
+it's ``false`` or missing, it has no effect.
+
+For example, let's say you have a source code repository ``foo``, which has
+some submodules, and you want ``west update`` to keep them in sync, along with
+another project named ``bar`` in the same workspace.
+
+You can do that with this manifest file:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         submodules: true
+       - name: bar
+
+Here, ``west update`` will initialize and update all submodules in ``foo``. If
+``bar`` has any submodules, they are ignored, because ``bar`` does not have a
+``submodules`` value.
+
+Option 2: List of mappings
+==========================
+
+The ``submodules`` key may be a list of mappings. Each element in the list
+defines the name of the submodule to update, and the path -- relative to the
+project's absolute path in the workspace -- to use for the submodule.
+
+For example, let's say you have a source code repository ``foo``, which has
+many submodules, and you want ``west update`` to just keep one of them in sync,
+along with another project named ``bar`` in the same workspace.
+
+You can do that with this manifest file:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: foo
+         submodules:
```

Comment:
I would recommend adding one more submodule in order to demonstrate the list having multiple (at least two) elements.
For example:
```
- name: foo
   submodules:
      - name: foo-first-sub
        path: path/of/first_submodule/you/want
      - name: foo-second-sub
        path: path/of/second_submodule/you/want
- name: bar
```

Reply:
I got to think.

Should we create an enhancement task here ?
What if `foo-first-sub` also contains submodules that I want, but I still don't want to enable all submodules for `foo`.

Reply:
The current implementation will update `foo-first-sub` and all of its submodules recursively, so I guess that addresses your question. Another question could be - what if `foo-first-sub` also contains submodules and I don't want to update them, but only the `foo-first-sub`. The second use case is currently not possible to do, as all submodules are updated recursively.

Reply:
I think someone who has complex submodules need should just disable them in west. Submodules are complex enough, if/when I have to actually deal with submodules the last thing I'd want is an incomplete layer of indirection on top of them. I see this new feature as a basic, simple "download all the pieces" feature for the users who do NOT have to touch submodules and don't even want to know about them.

Reply:
> The current implementation will update foo-first-sub and all of its submodules recursively, 

Then that is a very important piece of information that is missing.